File: src/main/java/it/contrader/dto/DocumentDTO.java
package it.contrader.dto;

import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DocumentDTO {
	
	private Long id;

	private String title;

	private String description;

	private String genre;
	
	private Date datePub;
	
	private UserDTO userDTO;
	
	private FolderDTO folderDTO;
	
	private CategoryDTO categoryDTO;
	
	private ExtensionDTO extensionDTO;

}

File: src/dto/documentdto.ts
import { CategoryDTO } from './categorydto';
import { ExtensionDTO } from './extensiondto';
import { FolderDTO } from './folderdto';
import { UserDTO } from './userdto';

export class DocumentDTO {
    id: number;
    title: string;
    description: string;
    genre: string;
    datePub: Date;
    user: UserDTO;
    folder: FolderDTO;
    category: CategoryDTO;
    extension: ExtensionDTO;
}
